lib/money_oxr/rates_store.rb

```diff
@@ -40,11 +40,11 @@ def loaded?
 
     def load
       # Loads data and ensures it is not stale.
-      if !loaded? && cache_path && File.exist?(cache_path)
-        load_from_cache_path
-      end
-      if app_id && (!loaded? || stale?)
-        load_from_api
+      return if loaded? && !stale?
+
+      with_cache_file_lock do
```

Comment:
Since load_from_cache_path and write_cache_file are public methods, maybe we want to move the `with_cache_file_lock` method call to inside those methods.  Then this could be simply `load_from_cache_path || load_from_api`.  What do you think?

Reply:
> Since load_from_cache_path and write_cache_file are public methods, maybe we want to move the with_cache_file_lock method call to inside those methods.

Large methods are not too good. But I don't see RuboCop in your project, and I've started to unwrap `with_cache_file_lock` method. But do you see double `yield` usage inside it? So it'd be hard, I guess. The separated method is OK.

> Then this could be simply load_from_cache_path || load_from_api. 

I think it's mistake: we're calling `load_from_cache_path` only for specific conditions and `load_from_api` either for other conditions.